api/server/router/image/image_routes.go

```diff
@@ -235,7 +235,11 @@ func (s *imageRouter) getImagesJSON(ctx context.Context, w http.ResponseWriter,
 		}
 	}
 
-	images, err := s.backend.Images(imageFilters, httputils.BoolValue(r, "all"), false)
+	images, err := s.backend.Images(ctx, types.ImageListOptions{
```

Comment:
The `shared-size` option should probably be gated by API version (and be ignored by `API < v1.41`

Perhaps here we should create a `opts := types.ImageListOptions{}` at the start of this function, then fill in options that are supported (possibly getting rid of the intermediate `imageFilters` variable)

Reply:
Something like this https://github.com/moby/moby/pull/42531/commits/7821e787813d356e018bd189e59a143b69e0516f ?